src/config/authd-config.c

```diff
@@ -192,4 +216,99 @@ short eval_bool(const char *str) {
         return OS_INVALID;
     }
 }
-#endif
\ No newline at end of file
+
+int get_time_interval(char *source, time_t *interval) {
+    char *endptr;
+    *interval = strtoul(source, &endptr, 0);
+
+    if ((!*interval && endptr == source) || *interval < 0) {
+        return OS_INVALID;
+    }
+
+    switch (*endptr) {
+    case 'd':
+        *interval *= 86400;
+        break;
+    case 'h':
+        *interval *= 3600;
+        break;
+    case 'm':
+        *interval *= 60;
+        break;
+    case 's':
+    case '\0':
+        break;
+    default:
+        return OS_INVALID;
+    }
+
+    return 0;
+}
+
+int w_read_force_config(XML_NODE node, authd_config_t *config) {
+    /* XML Definitions */
+    static const char *xml_enabled = "enabled";
+    static const char *xml_key_mismatch = "key_mismatch";
+    static const char *xml_disconnected_time = "disconnected_time";
+    static const char *xml_after_registration_time = "after_registration_time";
+
+    for (int i = 0; node[i]; i++) {
+        // enabled
+        if (!strcmp(node[i]->element, xml_enabled)) {
+            short b = eval_bool(node[i]->content);
+
+            if (b < 0) {
+                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            }
+
+            config->force_options.enabled = (b ? true : false);
+        }
+        // key_mismatch
+        else if (!strcmp(node[i]->element, xml_key_mismatch)) {
+            short b = eval_bool(node[i]->content);
+
+            if (b < 0) {
+                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            }
+
+            config->force_options.key_mismatch = (b ? true : false);
+        }
+        // disconnected_time
+        else if (!strcmp(node[i]->element, xml_disconnected_time)) {
+            if (node[i]->attributes && node[i]->attributes[0] && !strcmp(node[i]->attributes[0], xml_enabled)) {
```

Comment:
We could discuss this with the team, but I think the lack of **attribute** has to be considered as an error.

Reply:
One thing that was even worse, is that I was not validating the values. It could be different than null but with some random values and it will not throw any error about that. 
Let me know if something like this is better.

Without attribute
![2](https://user-images.githubusercontent.com/13010397/133120509-52168153-df32-4ffc-a83e-93ea5b0c6d66.png)
Attribute "enable" instead of "enabled"
![4](https://user-images.githubusercontent.com/13010397/133120510-c9ed850b-8a32-4cd5-a090-5ea367efec13.png)
Any random string instead of "yes" or "no"
![6](https://user-images.githubusercontent.com/13010397/133120513-92111a2b-1c1a-41a1-a8d4-83292a0eb47f.png)

Reply:
![1](https://user-images.githubusercontent.com/13010397/133120961-983379ef-a6e1-4d92-a697-aaeeb15d972d.png)

Reply:
Nice catch then !
It's more robust now